src/marqo/_httprequests.py

```diff
@@ -10,90 +10,71 @@
     BackendTimeoutError
 )
 
-s = requests.Session()
```

Comment:
There shouldn't be any impact if the `Session()` object used for sending HTTP requests is the same instantiated object in the `HttpRequests` class. As long as the `HttpRequests()` object initally created is the same each time we send a request via the `self_request` operation, it is effectively using the same `Session` object.

If a new `Session` object is created each time, we are basically making a new connection to the upstream server, **which we do not want** (as to avoid possible 503s due to TCP RST packet conflict)

Reply:
Thanks @VitusAcabado . Looking into this, it seems that it is best to keep this global. Especially when we consider the following use case:
```
mq = marqo.Client()
mq.index('my-index').add_documents([{"Title": "Hello world"}])
mq.index('my-index').search('hello')
```
Line 2,3 both instantiate an Index Object, which in turn instantiates an HTTPRequest so there is no optimisation to be had. There would be a new HTTPRequest, and therefore session object, per line. See here for reference: https://github.com/marqo-ai/py-marqo/blob/mainline/src/marqo/client.py#L107

Reply:
This is a good spot. I think `def __init__(self, config: Config, s: requests.session = session) -> None:` will solve the issue (i.e. we can pass in session for testing, and use the global variable `session` more regularly).